apollo-router/src/axum_factory/compression/mod.rs

```diff
@@ -123,19 +124,25 @@ where {
                 }
             }
 
-            let buf = BytesMut::zeroed(64);
-            let mut partial_output = PartialBuffer::new(buf);
+            loop {
+                let buf = BytesMut::zeroed(1024);
+                let mut partial_output = PartialBuffer::new(buf);
 
-            match self.finish(&mut partial_output) {
-                Err(e) => {
-                    let _ = tx.send(Err(e.into())).await;
-                }
-                Ok(_) => {
-                    let len = partial_output.written().len();
-
-                    let mut buf = partial_output.into_inner();
-                    buf.resize(len, 0);
-                    let _ = tx.send(Ok(buf.freeze())).await;
+                match self.finish(&mut partial_output) {
+                    Err(e) => {
+                        let _ = tx.send(Err(e.into())).await;
+                        break;
+                    }
+                    Ok(is_flushed) => {
+                        let len = partial_output.written().len();
+
+                        let mut buf = partial_output.into_inner();
+                        buf.resize(len, 0);
+                        let _ = tx.send(Ok(buf.freeze())).await;
```

Comment:
Maybe break if there is an error here

Reply:
fixed in [7a8eda9](https://github.com/apollographql/router/pull/3037/commits/7a8eda97f89d6de41499066a118152f3ff521a98)